app/lib/features/attachments/widgets/attachment_item.dart

```diff
@@ -1,41 +1,188 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:share_plus/share_plus.dart';
 
 // Attachment item UI
 class AttachmentItem extends ConsumerWidget {
   final Attachment attachment;
+  final bool canEdit;
+
   // whether item can be viewed on gesture
   final bool? openView;
+
   const AttachmentItem({
     super.key,
     required this.attachment,
+    this.canEdit = false,
     this.openView = true,
   });
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    var msgContent = attachment.msgContent();
-    String type = attachment.typeStr();
-    if (type == 'image') {
-      return AttachmentContainer(
-        name: msgContent.body(),
-        child: ImageView(attachment: attachment, openView: openView),
-      );
-    } else if (type == 'video') {
-      return AttachmentContainer(
-        name: msgContent.body(),
-        child: VideoView(attachment: attachment, openView: openView),
-      );
+    final containerColor = Theme.of(context).colorScheme.surface;
+    final borderColor = Theme.of(context).colorScheme.primary;
+    final attachmentType = AttachmentType.values.byName(attachment.typeStr());
+    final eventId = attachment.attachmentIdStr();
+    final roomId = attachment.roomIdStr();
+    final mediaState = ref.watch(attachmentMediaStateProvider(attachment));
+
+    return Container(
+      decoration: BoxDecoration(
+        color: containerColor,
+        borderRadius: BorderRadius.circular(8),
+        border: Border.all(color: borderColor),
+      ),
+      child: ListTile(
+        leading:
+            mediaState.mediaLoadingState.isLoading || mediaState.isDownloading
+                ? loadingIndication()
+                : attachmentLeadingIcon(attachmentType),
+        onTap: () => attachmentOnTap(
+          ref,
+          context,
+          attachmentType,
+          mediaState,
+        ),
+        onLongPress: canEdit
+            ? () => openRedactContentDialog(
+                  context,
+                  eventId: eventId,
+                  roomId: roomId,
+                  title: L10n.of(context).deleteAttachment,
+                  description: L10n.of(context)
+                      .areYouSureYouWantToRemoveAttachmentFromPin,
+                  isSpace: true,
+                )
+            : null,
+        title: title(attachmentType),
+        trailing: Visibility(
+          visible: mediaState.mediaFile == null &&
+              attachmentType != AttachmentType.link,
+          child: IconButton(
+            onPressed: () => attachmentOnTap(
+              ref,
+              context,
+              attachmentType,
+              mediaState,
+            ),
+            icon: const Icon(Atlas.download_arrow_down),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget title(AttachmentType attachmentType) {
+    final msgContent = attachment.msgContent();
```

Comment:
there is also `final title = attachment.name()` . I think this is what you want to use and why you said that the title attributed isn't coming through. It is new but the thing you want to use!

Reply:
`attachment.name()` want't giving me data somehow so used other way to handle it for temp. But yes this is not proper way and have this in my plan it use name() or relevant field. Right after this PR I am planning to fix this stuff.